### PR TITLE
Expect/side bar

### PR DESCRIPTION
**Pull request by leomendoza123**
Expectations based on sketch designs for https://trello.com/c/Qej43BsL/6929-modal-windows-my-orcid-components-8 and https://trello.com/c/q4FAfJAx

### CODE REVIEW

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      // TODO @dandineen no sketch that displays the input box
+    })
+    it('remove/delete  emails and display those as "unverified" with default privacy', () => {
+      // Expect changes to be display outside and inside of the modal
+    })
+    it('display emails "unverified" and "verified"', () => {
+      // Expect changes to be display outside and inside of the modal
+    })
+    it('resend the email verification email', () => {
+      // TODO @dandineen no button to resend
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:20]: I've added in a 'Verify email' link, sitting at the end of the field itself. Truth be told I'm not 100% sure about it. I may need to give the email form further thought to see if I can make something a little more elegant.

https://zpl.io/VYOlJqe

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    it('change the email privacy', () => {})
+    it('make changes and cancel', () => {
+      // Expect changes NOT to be display outside and inside of the modal
+    })
+    it('clicks outside and DONT close the modal ', () => {})
+    it('open the terms of use on a separate window ', () => {})
+  })
+  describe('Websites and socials links', () => {
+    it('display a user with no items', () => {
+      // TODO @dandineen should this look just empty on the modal
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:35]: I've added designs and example copy for the 'empty' lists in the modal forms -> https://zpl.io/b6qnERg

It has raised a question about the possibility of changing the 'Add another X' link text when the list is empty. In that case we really want the link to read `Add an X` instead. Is that something that can be done without too much difficulty?

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      // Expect changes to be display outside and inside of the modal
+    })
+    it('make changes and cancel', () => {
+      // Expect changes NOT to be display outside and inside of the modal
+    })
+    it('clicks outside and DONT close the modal ', () => {})
+  })
+  describe('Countries', () => {
+    it('display a user with no items', () => {
+      // TODO @dandineen should this look just empty on the modal
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:57]: See comment on line 35

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    })
+    it('make changes and cancel', () => {
+      // Expect changes NOT to be display outside and inside of the modal
+    })
+    it('clicks outside and DONT close the modal ', () => {})
+  })
+  describe('Other IDs', () => {
+    it('display a user with no items', () => {
+      // should be hidden from the side bar https://trello.com/c/JZJ75TWl/35-how-to-add-other-ids-is-confusing
+      // TODO @dandineen should this look just empty on the modal
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:80]: See comment on Line 35

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      // Expect changes to be display outside and inside of the modal
+    })
+    it('make changes and cancel', () => {
+      // Expect changes NOT to be display outside and inside of the modal
+    })
+    it('clicks outside and DONT close the modal ', () => {})
+  })
+  describe('Keywords', () => {
+    it('display a user with no items', () => {
+      // TODO @dandineen should this look just empty on the modal
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:99]: See comment on line 35

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    })
+    it('clicks outside and DONT close the modal ', () => {})
+  })
+  describe('Countries', () => {
+    it('display a user with no items', () => {
+      // TODO @dandineen should this look just empty on the modal
+    })
+    it('add items and display those with default privacy', () => {
+      // Expect changes to be display outside and inside of the modal
+      // TODO @dandineen no sketch that displays the input box
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:61]: Example added in Zeplin -> https://zpl.io/2jGAl3r

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+    })
+    it('clicks outside and DONT close the modal ', () => {})
+  })
+  describe('Keywords', () => {
+    it('display a user with no items', () => {
+      // TODO @dandineen should this look just empty on the modal
+    })
+    it('add items and display those with default privacy', () => {
+      // Expect changes to be display outside and inside of the modal
+      // TODO @dandineen no sketch that displays the input box
```
Comment (dandineen) [inline cypress/integration/my-orcid-sidebar.e2e.js:103]: Example added to Zeplin -> https://zpl.io/brwdkA5

**Review by dandineen — COMMENTED** [review]
Hey @leomendoza123! This might not have been what you were expecting for this doc review. If not please give me a shout and I'll re-format in whatever way suits you best.

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+  describe('Orcid id', () => {
+    it('display the users Orcid', () => {})
+    it('display url to navigate to the public page view', () => {})
+  })
+  describe('Emails' + runInfo(), () => {
+    it('display a user with only a primary email "unverified"', () => {})
+    it('display a user with only a primary email "verified"', () => {})
+    it('add additional emails and display those as "unverified" with default privacy', () => {
+      // Expect changes to be display outside and inside of the modal
+      // TODO @dandineen no sketch that displays the input box
```
[the remark this answers is not in the document]
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:11]: Yes, this is true! 
This sidebar document is including the modals which might as well have their own test document. 
But let's keep it as it is for now unless it becomes too painful.

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 2 above)
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:35]: There would be no problem, we can change the string in the button depending on any condition. Please just update that as you wish on the designs

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    it('display the users Orcid', () => {})
+    it('display url to navigate to the public page view', () => {})
+  })
+  describe('Emails' + runInfo(), () => {
+    it('display a user with only a primary email "unverified"', () => {})
+    it('display a user with only a primary email "verified"', () => {})
+    it('display multiple emails "unverified" and "verified"', () => {
+      // Expect changes to be display outside and inside of the modal
+    })
+    describe('Emails edit modals' + runInfo(), () => {
```
Comment (leomendoza123) [inline cypress/integration/my-orcid-sidebar.e2e.js:12]: Should the email list changes be saved right after those are done by the user (as it currently works) or only after the user clicks the save button (as most other modals currently work)?.

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+      it('add additional emails and display those as "unverified" with default privacy', () => {
+        // Expect changes to be display outside and inside of the modal
+      })
+      it('remove emails', () => {
```
Comment (leomendoza123) [inline cypress/integration/my-orcid-sidebar.e2e.js:16]: Should we show an alert message right after an email delete button is pressed.

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+        // Expect the primary email to not display the delete button
+        // Expect changes to be display outside and inside of the modal
+      })
+      it('edit a non-primary and display it as "unverified"', () => {
+        // Expect changes to be display outside and inside of the modal
+      })
+      it('edit a primary email and display it as "verified"', () => {
```
Comment (leomendoza123) [inline cypress/integration/my-orcid-sidebar.e2e.js:23]: Should we prompt a "please verify your primary email" type of message after a primary email is edited? as the current app does

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+        // Expect changes to be display outside and inside of the modal
+      })
+      it('display multiple emails "unverified" and "verified"', () => {
+        // Expect changes to be display outside and inside of the modal
+      })
+      it('resend the email verification email', () => {})
```
Comment (leomendoza123) [inline cypress/integration/my-orcid-sidebar.e2e.js:29]: Should we show an alert message right after a verification email is resend?

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+  })
+  describe('Emails' + runInfo(), () => {
+    it('display a user with only a primary email "unverified"', () => {})
+    it('display a user with only a primary email "verified"', () => {})
+    it('display multiple emails "unverified" and "verified"', () => {
+      // Expect changes to be display outside and inside of the modal
+    })
+    describe('Emails edit modals' + runInfo(), () => {
+      it('add additional emails and display those as "unverified" with default privacy', () => {
+        // Expect changes to be display outside and inside of the modal
```
Comment (leomendoza123) [inline cypress/integration/my-orcid-sidebar.e2e.js:14]: After the user clicks the "Add another email" URL at the bottom, a new empty input box is shown. 
while the new input box is empty or invalid should the "Add another email" URL be hidden, disable, or kept?

File: cypress/integration/my-orcid-sidebar.e2e.js [hunk 14]
```diff
@@ -0,0 +1,118 @@
+describe('My Orcid sidebar' + runInfo(), () => {
+  describe('Orcid id', () => {
+    it('display the users Orcid', () => {})
+    it('display url to navigate to the public page view', () => {})
+  })
+  describe('Emails' + runInfo(), () => {
+    it('display a user with only a primary email "unverified"', () => {})
+    it('display a user with only a primary email "verified"', () => {})
+    it('display multiple emails "unverified" and "verified"', () => {
```
Comment (leomendoza123) [inline cypress/integration/my-orcid-sidebar.e2e.js:9]: Is the `bold` font on the emails meant for the primary email? just making sure

Comment (dandineen) [reply cypress/integration/my-orcid-sidebar.e2e.js:9]: That's right. In the sidebar the primary email is set in bold to make it stand out a little mroe than the other email addresses.

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 13 above)
Comment (dandineen) [reply cypress/integration/my-orcid-sidebar.e2e.js:14]: I'd keep it there and active. If the user wishes to add multiple empty emails then they can. I'm assuming empty or invalid fields wouldn't be saved or would show an error on 'Save changes'?

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 10 above)
Comment (dandineen) [reply cypress/integration/my-orcid-sidebar.e2e.js:16]: I'm inclined to say 'No' to this as it just seems like overkill. Are secondary emails really _so_ important that a user needs to be reminded they've just clicked the button to delete one?

If we keep things consistent behaviour-wise then as soon as a user clicks the delete button on a secondary email address it'll be removed from the list. It literally disappears! However, it's not actually deleted until the user confirms the action by clicking 'Save changes'. Cancelling by clicking outside of the modal, clicking the [X] button or pressing 'Cancel' will undo these changes.

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 13 above)
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:14]: Yes we can handle that a similar thing to the "additional emails" on the register form

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 10 above)
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:16]: Yes!, I don't think we need to ask the users to confirm the email deletion if now the email deleting will only happen after the save button is pressed.  

I will keep this comment as `unresolved` for now until we have these scenarios well defined.

**Comment by leomendoza123** [comment]
[[speedup] Bio: Emails (5)](https://trello.com/c/4EOdjRy6/7214-speedup-bio-emails-5)

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 10 above)
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:16]: As we have concluded at previous meetings, we will not show this deleted email message.

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 12 above)
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:29]: As we have concluded, yes an alert message is show right after, as show on the new sketches zpl://screen?pid=5ee88fa6b1d4aa97af37a43a&sid=60085d04156ab30330554f3a

File: cypress/integration/my-orcid-sidebar.e2e.js (hunk 11 above)
Comment (leomendoza123) [reply cypress/integration/my-orcid-sidebar.e2e.js:23]: As concluded on previous meetings, edit the primary email won't be allow

**Pull request merged**